speclang_modern.tex

```diff
@@ -2127,6 +2121,97 @@ \subsection{Non-terminating functions}
 \listinginput{1}{terminates_list.c}
 \end{example}
 
+\paragraph{Semantics in statement contract}
+The termination of statements can also be specified in statement contracts in using
+the same clause.
+When such a clause is omitted the statement contract does not add any more constraints
+to the termination of the related statement (and, by the way, to the statements that it contains).
+
+The semantics of a clause \lstinline|terminates| \lstinline|T| of a contract of
+a statement \lstinline|S| is defined inductively in considering the
+statement \lstinline|Sb| that is the innermost block (or compound statement such
+as \lstinline|if|, ...) that has a contract with a clause \lstinline|terminates|
+\lstinline|Tb| and encloses \lstinline|S|.
+This search is performed recursively, until a statement \lstinline|Sb| is found
+(the body of the function can be considered for that and in such a case, the
+\lstinline|terminates| clause of the function contract gives \lstinline|Tb|).
+
+\begin{itemize}
+  \item If \lstinline|Tb| holds in the pre-state of \lstinline|Sb|\footnote{In
+        this definition \lstinline|Sb| denotes a statement as much as a function
+        body.} then \lstinline|T| must holds in the pre-state of \lstinline|S|
+        (at the beginning of each execution of \lstinline|S|).
+  \item The statement \lstinline|S| must terminate when \lstinline|T| holds in
+        its pre-state\footnote{It must terminate regardless of any other
+        \lstinline|terminates| clause provided for a block (or a function) that
+        contains it.}.
+\end{itemize}
+
+By transitivity and induction, this definition implies all statements of
```

Comment:
Should be _... implies **that** all statements..._

Reply:
```suggestion
By transitivity and induction, this definition implies that all statements of
```